src/tests/Interop/MarshalAPI/FunctionPointer/FunctionPointer.cs

```diff
@@ -111,4 +112,81 @@ public static void RunGetDelForOutIntTest()
         }
         Assert.Equal(expectedValue, outVar);
     }
+
+    [UnmanagedCallersOnly(CallConvs = [typeof(CallConvCdecl)])]
+    static int UnmanagedExportedFunction(float arg)
+    {
+        return Convert.ToInt32(arg);
+    }
+
+    class GenericCaller<T>
+    {
+        internal static unsafe T GenericCalli<U>(void* fnptr, U arg)
+        {
+            return ((delegate* unmanaged[Cdecl]<U, T>)fnptr)(arg);
+        }
+
+        internal static unsafe BlittableGeneric<T> WrappedGenericCalli<U>(void* fnptr, U arg)
+        {
+            return ((delegate* unmanaged[Cdecl]<U, BlittableGeneric<T>>)fnptr)(arg);
+        }
+    }
+
+    struct BlittableGeneric<T>
+    {
+        public int X;
+    }
+
+    [Fact]
+    public static void RunGenericFunctionPointerTest()
+    {
+        Console.WriteLine($"Running {nameof(RunGenericFunctionPointerTest)}...");
+        int outVar = 0;
+        int expectedValue = 42;
+
+        unsafe
+        {
+            outVar = GenericCaller<int>.GenericCalli((delegate* unmanaged[Cdecl]<float, int>)&UnmanagedExportedFunction, 42.0f);
+        }
+        Assert.Equal(expectedValue, outVar);
+        
+        outVar = 0;
+        unsafe
+        {
+            outVar = GenericCaller<int>.WrappedGenericCalli((delegate* unmanaged[Cdecl]<float, int>)&UnmanagedExportedFunction, 42.0f).X;
+        }
+        Assert.Equal(expectedValue, outVar);
+
+        outVar = 0;
+        unsafe
+        {
+            outVar = GenericCaller<string>.WrappedGenericCalli((delegate* unmanaged[Cdecl]<float, int>)&UnmanagedExportedFunction, 42.0f).X;
+        }
+        Assert.Equal(expectedValue, outVar);
+    }
+
+    [Fact]
+    public static void RunInvalidGenericFunctionPointerTest()
+    {
+        Console.WriteLine($"Running {nameof(RunInvalidGenericFunctionPointerTest)}...");
+        unsafe
+        {
+            nint lib = nint.Zero;
+            try
+            {
+                lib = NativeLibrary.Load(nameof(FunctionPointerNative));
+                nint fnptr = NativeLibrary.GetExport(lib, "ReturnParameter");
```

Comment:
This is fetching method with default calling convention, but casting it to the CDecl calling convention. The calling conventions should match.

Reply:
I have removed the `cdecl` here and use default calling conv instead.